pagopa/io-app
# feat: [IOCOM-1665] FIMS Authentication proper UI

⚠️ This PR depends on #6330 ⚠️

## Short description
This PR implements the UI for a FIMS Authentication error.

|iOS<br />SSO error|iOS<br />Generic error|
|-|-|
|![iOSOne](https://github.com/user-attachments/assets/1806520f-b319-4b71-a7dd-d19a9e258e21)|![iOSTwo](https://github.com/user-attachments/assets/941e4e40-ccbd-4640-8553-94f9220de523)|

|Android<br />SSO error|Android<br />Generic error|
|-|-|
|![AndroidOne](https://github.com/user-attachments/assets/1e3afb6e-c81d-4784-8943-cdf07f497c09)|![AndroidTwo](https://github.com/user-attachments/assets/605f46ec-f06f-455d-a322-eeb2ebe46770)|

## List of changes proposed in this pull request
- `handleFimsGetConsentsList` refactored to distinguish between a generic error and a specific SSO one
- `FimsFullScreenErrors` refactored to retrieve error type and error content on its own. It also uses the `useDebugInfo` hook to display debug data when such mode is on
- `FIMSFlowHandlerScreen` refactored to detect an error and return the `FimsFullScreenErrors` component, instead of passing the specific error to the latter
- `AUTHENTICATION` tag added to `FIMS_SSO_ERROR_TAGS`
- `DEBUG` tag removed from `FIMS_SSO_ERROR_TAGS` (it is now handled by the `useDebugInfo` hook, combined with the new `fimsDebugDataSelector`)
- New `fimsAuthenticationFailedSelector` and `fimsAuthenticationErrorTagSelector` to know if there was an error and to retrieve its category
- Removed deprecated `standardMessage` data and its selector/usages
- `consentsData` renamed to `ssoData` since it is also used after consents have been retrieved

## How to test
Using the io-dev-api-server, modify the Relying Party redirect to send invalid parameters (like the `clientId`, `scopes` and so on). Check that the proper error screen is displayed. To test the generic error, change any other flow (starting from the consents list sending) to return a 500 http status error code.
